DFC.ServiceTaxonomy.GraphVisualiser/Controllers/VisualiseController.cs

```diff
@@ -0,0 +1,103 @@
+// or check out all existing owl types and see if can piggy-back or build on any
+
+// (b) e.g. (software developer:jobProfile)--(Tasks)--(coding)
+
+//todo: move visualisation into GraphSync module? then could add view graph button to sync part (along with id)
+//todo: on properties pop.up have edit button which links to content edit page
+
+//todo: maxLabelWidth to 180 : need to add options support to import json
+
+namespace DFC.ServiceTaxonomy.GraphVisualiser.Controllers
+{
+    public class VisualiseController : Controller
+    {
+        private readonly IGraphDatabase _neoGraphDatabase;
+        private readonly IContentDefinitionManager ContentDefinitionManager;
+        private readonly INeo4JToOwlGeneratorService Neo4JToOwlGeneratorService;
+        private readonly IOrchardToOwlGeneratorService OrchardToOwlGeneratorService;
+
+        private readonly JsonSerializerOptions JsonOptions = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+        };
+
+        public VisualiseController(IGraphDatabase neoGraphDatabase, IContentDefinitionManager contentDefinitionManager, INeo4JToOwlGeneratorService neo4jToOwlGeneratorService, IOrchardToOwlGeneratorService orchardToOwlGeneratorService)
+        {
+            _neoGraphDatabase = neoGraphDatabase ?? throw new ArgumentNullException(nameof(neoGraphDatabase));
+            ContentDefinitionManager = contentDefinitionManager ?? throw new ArgumentNullException(nameof(contentDefinitionManager));
+            Neo4JToOwlGeneratorService = neo4jToOwlGeneratorService ?? throw new ArgumentNullException(nameof(neo4jToOwlGeneratorService));
+            OrchardToOwlGeneratorService = orchardToOwlGeneratorService ?? throw new ArgumentNullException(nameof(orchardToOwlGeneratorService));
+        }
+
+        public ActionResult Viewer()
+        {
+            return View();
+        }
+
+        public async Task<ActionResult> Data([FromQuery] string uri)
+        {
+            if (string.IsNullOrWhiteSpace(uri))
+            {
+                return await GetOntology();
+            }
+            else
+            {
+                return await GetData(uri);
+            }
+        }
+
+        public async Task<ActionResult> GetOntology()
+        {
+            var contentTypeDefinitions = ContentDefinitionManager.ListTypeDefinitions();
+
+            var owlDataModel = OrchardToOwlGeneratorService.CreateOwlDataModels(contentTypeDefinitions);
+            var owlResponseString = JsonSerializer.Serialize(owlDataModel, JsonOptions);
+
+            return Content(owlResponseString, MediaTypeNames.Application.Json);
+        }
+
+        public async Task<ActionResult> GetData(string uri)
+        {
+            const string prefLabel = "skos__prefLabel";
+            var query = new GetNodesCypherQuery(nameof(uri), uri, prefLabel, prefLabel);
+
+            _ = await _neoGraphDatabase.Run(query);
```

Comment:
What is _? This method doesn't seem to do anything with the result

Reply:
_ is a discard, yes it doesn't use the records